addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,218 @@
+        dest_loc = moves.mapped("location_dest_id")
+
+        group = ProcurementGroup.get_or_create(group_key, create=True)
+
+        # Look for an existing picking with the right group
+        picking = self.search(
+            [
+                ("picking_type_id", "=", picking_type.id),
+                ("location_id", "=", src_loc.id),
+                ("location_dest_id", "=", dest_loc.id),
+                ("group_id", "=", group.id),
+                # NB: only workable pickings
+                ("state", "in", ["assigned", "confirmed", "waiting"]),
+            ]
+        )
+
+        if not picking:
+            # Otherwise reuse the existing picking if all the moves
+            # already belong to it and it contains no other moves
+            # The picking_type_id, location_id and location_dest_id
+            # will match already
+            current_picking = moves.mapped("picking_id")
+            if len(current_picking) == 1 and current_picking.mapped("move_lines") == moves:
+                values = {"group_id": group.id}
+                values.update(kwargs)
+
+                current_picking.write(values)
+                picking = current_picking
+
+        if not picking or len(picking) > 1:
+            # There was no suitable picking to reuse.
+            # Create a new picking.
+            picking_values = self._get_default_new_picking_for_group_values(
+                group, picking_type, src_loc, dest_loc
+            )
+            picking_values.update(kwargs)
+
+            picking = self.create_picking(picking_type, **picking_values)
+
+        else:
+            picking._remove_misleading_values(**kwargs)
+
+        moves.write({"group_id": group.id, "picking_id": picking.id})
+
+        move_lines = moves.mapped("move_line_ids")
+        if move_lines:
+            move_lines.write({"picking_id": picking.id})
+            # After moving move lines check entire packages again just in case
+            # some of the move lines are completing packages
+            if picking.state != "done":
+                picking._check_entire_pack()
+
+        return picking
+
+    def _prepare_extra_info_for_new_picking_for_group(self, moves):
+        """
+        Given the group of moves to refactor and its related pickings,
+        prepare the extra info for the new pickings that might be created
+        for the group of moves.
+        Fields with more than one value are going to be ignored.
+        """
+        values = {}
+
+        origins = list(set(self.mapped("origin")))
+        if len(origins) == 1:
+            values["origin"] = origins[0]
+
+        partners = self.partner_id
+        if len(partners) == 1:
+            values["partner_id"] = partners.id
+
+        dates_done = list(set(moves.mapped("date")))
+        if len(dates_done) == 1:
+            values["date_done"] = dates_done[0]
+
+        return values
+
+    def _get_package_search_domain(self, package):
```

Comment:
Is this still needed?

Reply:
Nope, removed. :) Was previously called by `get_pickings` which was removed in an earlier commit.